Assets/_Laz/Editor/PatrolLocationHandler.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using UnityEditor;
+using Laz;
+using UnityEngine;
+
+
+namespace LazEditor
+{
+    [CustomEditor(typeof(PatrolBehaviour))]
+    public class PatrolLocationHandler : Editor
+    {
+        private const float DRAW_PLANE_HEIGHT = 0f;
+        private const float CLOSEST_DISTANCE_FOR_DELETION = 0.6f;
+        private PatrolBehaviour _behaviour = null;
+        private Vector3 _tempPosition = Vector3.zero;
+        private List<Vector3> _patrolTrail;
+
+        private void OnEnable()
+        {
+            _behaviour = (PatrolBehaviour) target;
+        }
+
+        private void OnSceneGUI()
+        {
+            if (_behaviour == null)
+            {
+                return;
+            }
+            
+            Event guiEvent = Event.current;
+            
+            HandleInput(guiEvent);
+            HandlePatrolPositioning();
+        }
+
+        private void HandlePatrolPositioning()
+        {
+            for (int i = 0, count = _behaviour.PatrolPositions.Count; i < count; i++)
+            {
+                _tempPosition = Handles.PositionHandle(_behaviour.PatrolPositions[i], Quaternion.identity);
+                _tempPosition.y = 0.0f; // Ensure the position is grounded onto the plane
+
+                if (!_behaviour.PatrolPositions[i].Equals(_tempPosition))
+                {
+                    Undo.RecordObject(_behaviour, "Patrol Positioning");
+                    _behaviour.PatrolPositions[i] = _tempPosition;   
+                }
+            }
+            
+            _patrolTrail = new List<Vector3>();
+            _patrolTrail.Add(_behaviour.transform.position);
+            _patrolTrail.AddRange(_behaviour.PatrolPositions);
+            
+            Handles.color = Color.yellow;
+            Handles.DrawPolyLine(_patrolTrail.ToArray());
+        }
+
+        private void HandleInput(Event guiEvent)
+        {
+            if (guiEvent.type != EventType.MouseDown)
```

Comment:
Is it possible break this method into smaller chunks? 

Like have a method just for the mouse,
have a method just for "Ctrl/Cmd"
have a method just for "Shift"...ect